pymc_marketing/clv/models/shifted_beta_geo.py

```diff
@@ -77,59 +76,81 @@ class ShiftedBetaGeoModelIndividual(CLVModel):
            https://journals.sagepub.com/doi/pdf/10.1002/dir.20074
     """
 
-    _model_name = "Shifted-Beta-Geometric Model (Individual Customers)"
+    _model_type = "Shifted-Beta-Geometric Model (Individual Customers)"
 
     def __init__(
         self,
-        customer_id: Union[np.ndarray, pd.Series],
-        t_churn: Union[np.ndarray, pd.Series],
-        T: Union[np.ndarray, pd.Series],
-        alpha_prior: Optional[TensorVariable] = None,
-        beta_prior: Optional[TensorVariable] = None,
+        data: pd.DataFrame,
+        model_config: Optional[Dict] = None,
+        sampler_config: Optional[Dict] = None,
     ):
-        super().__init__()
-
-        t_churn = np.asarray(t_churn)
-        T = np.asarray(T)
-
-        if np.any((t_churn < 0) | (t_churn > T) | np.isnan(t_churn)):
+        try:
+            self.customer_id = data["customer_id"]
+        except KeyError:
+            raise KeyError("data must contain a 'customer_id' column")
+        try:
+            self.t_churn: np.ndarray = np.asarray(data["t_churn"])
+        except KeyError:
+            raise KeyError("data must contain a 't_churn' column")
+        try:
+            self.T: np.ndarray = np.asarray(data["T"])
+        except KeyError:
+            raise KeyError("data must contain a 'T' column")
+        super().__init__(model_config=model_config, sampler_config=sampler_config)
+        self.data = data
+        self.alpha_prior = self.create_distribution_from_prior(
+            self.model_config["alpha_prior"]["dist"],
+            **self.model_config["alpha_prior"]["kwargs"]
+        )
+        self.beta_prior = self.create_distribution_from_prior(
+            self.model_config["beta_prior"]["dist"],
+            **self.model_config["beta_prior"]["kwargs"]
+        )
+        self._process_priors(self.alpha_prior, self.beta_prior)
+
+        if np.any(
+            (self.t_churn < 0) | (self.t_churn > self.T) | np.isnan(self.t_churn)
+        ):
             raise ValueError(
                 "t_churn must respect 0 < t_churn <= T.\n",
                 "Customers that are still alive should have t_churn = T",
             )
+        self.coords = {"customer_id": np.asarray(self.customer_id)}
 
-        alpha_prior, beta_prior = self._process_priors(alpha_prior, beta_prior)
+    @property
+    def default_model_config(self) -> Dict:
+        return {
+            "alpha_prior": {"dist": "halfflat", "kwargs": {}},
+            "beta_prior": {"dist": "halfflat", "kwargs": {}},
+        }
 
-        coords = {"customer_id": np.asarray(customer_id)}
-        with pm.Model(coords=coords) as self.model:
-            alpha = self.model.register_rv(alpha_prior, name="alpha")
-            beta = self.model.register_rv(beta_prior, name="beta")
+    @property
+    def _serializable_model_config(self) -> Dict:
```

Comment:
Why isn't this in the base CLVModel class?

Reply:
because technically every single model type can have it's own unserialisable parts of prior that would need to be converted to their serialisable versions here. If we're sure that we don't need it I can move it to basic.py

Reply:
We should only define it when needed. The default seems sensible enough